### PR TITLE
LABS-654-new: use fileProcess function to do our processes

### PR DESCRIPTION
**Pull request by SalehElnagarSecurrency**
Signed-off-by: SalehElnagarSecurrency <saleh.elnagar@securrency.com>

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -217,11 +224,35 @@ func processFile(
 		meta.Title = mark.ExtractDocumentLeadingH1(markdown)
 	}
 
-	if meta.Title == "" {
+	path := strings.Split(file, "/")
+	title := strings.Replace(path[len(path)-1], ".md", "", -1)
```
Comment (AlexeyShchukinSecurrency) [inline main.go]: I'd change it to something like:
```go
title := strings.Trim(filepath.Base(p), ".md")
```
https://play.golang.com/p/9MU02vwNnNg

File: main.go [hunk 2]
```diff
@@ -217,11 +224,35 @@ func processFile(
...
+
+	config, _ := LoadConfig(flags.Config)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	additional_parents := strings.Split(file, "/")
+	additional_parents = additional_parents[1 : len(additional_parents)-1]
+	for i := 0; i < len(additional_parents); i++ {
+		additional_parents[i] = fmt.Sprintf("%s-ADR", additional_parents[i])
```
Comment (AlexeyShchukinSecurrency) [inline main.go]: What if we want to sync something outside the ADR repository? To every title `-ADR` will be appended anyway. I suggest making it optional and configurable (e.g. via config file).

File: main.go [hunk 3]
```diff
@@ -217,11 +224,35 @@ func processFile(
...
+		meta.Title = strings.Title(title)
+	}
+
+	config, _ := LoadConfig(flags.Config)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	additional_parents := strings.Split(file, "/")
+	additional_parents = additional_parents[1 : len(additional_parents)-1]
```
Comment (AlexeyShchukinSecurrency) [inline main.go]:
```suggestion
	additional_parents := strings.Split(filepath.Dir(p), "/")
```

File: main.go (hunk 2 above)
Comment (AlexeyShchukinSecurrency) [inline main.go]: There is a syntax sugar for loops:
```go
for i, v := additional_parents {
	additional_parents[i] = fmt.Sprintf("%s-ADR", v)
```

**Review by AlexeyShchukinSecurrency — COMMENTED** [review]
Please check my comments.

File: main.go (hunk 1 above)
Comment (SalehElnagarSecurrency) [reply main.go]: done

File: main.go (hunk 2 above)
Comment (SalehElnagarSecurrency) [reply main.go]: done

Comment (SalehElnagarSecurrency) [reply main.go]: done

**Review by AlexeyShchukinSecurrency — APPROVED** [review]
Looks good to me.

**Pull request merged**